storybookjs/storybook
# Core: Simplified hierarchy separators

Issue: #8793 

## What I did

- Add warnings when separators are set.
- Change behaviour when hierarchy separators are not set, as per issue.
- Add showRoots option and change behaviour when set.

## How to test

- Added a test for `stories.js` and new behaviour -- should I check the deprecation message is called?

- Run official storybook to see deprecation message when kinds still have '/' or '.'
- Run other examples to see deprecation message when option is set
- Run `examples/cra-react15` to see new behaviour (try setting `showRoots: true`)

- Does this need an update to the documentation? Yes, I removed documentation about the old option, added documentation for the new option, and migration notes.

If your answer is yes to any of these, please make sure to include it in your PR.

## Still TODO (on this PR or later)

 - [x] Can we add a test for the docs part? @shilman can you help me here?
 - [x] Add a codemod to convert `.` and `|` to `/` in kinds.
 - [x] Update all our example apps to use the new API
 - [ ] Check all our documentation for uses of `|` or `.`
